### PR TITLE
Allow setting parcel when uploading a certificate of title

### PR DESCRIPTION
**Pull request by dhaselhan**

### CODE REVIEW

File: alcs-frontend/src/app/features/application/documents/document-upload-dialog/document-upload-dialog.component.ts [hunk 1]
```diff
@@ -98,7 +110,10 @@ export class DocumentUploadDialogComponent implements OnInit, OnDestroy {
       visibilityFlags,
     };
     this.isSaving = true;
-    if (this.data.existingDocument) {
+    if (parcelId && dto.file && this.type.value === DOCUMENT_TYPE.CERTIFICATE_OF_TITLE) {
+      // @ts-ignore File is checked above to not be undefined, typescript still thinks it might be undefined
+      await this.applicationDocumentService.attachCertificateOfTitle(this.data.fileId, parcelId, dto);
```
Comment (mhuseinov) [inline alcs-frontend/src/app/features/application/documents/document-upload-dialog/document-upload-dialog.component.ts:115]: Maybe this one will be better than ts-ignore 
this.data.fileId! 
or just check in if statement

Comment (dhaselhan) [reply alcs-frontend/src/app/features/application/documents/document-upload-dialog/document-upload-dialog.component.ts]: Cleaned up!

File: alcs-frontend/src/app/features/application/documents/document-upload-dialog/document-upload-dialog.component.ts [hunk 2]
```diff
@@ -127,12 +147,48 @@ export class DocumentUploadDialogComponent implements OnInit, OnDestroy {
     );
   }
 
-  onDocTypeSelected($event?: ApplicationDocumentTypeDto) {
+  async prepareCertificateOfTitleUpload(uuid?: string) {
+    const parcels = await this.parcelService.fetchParcels(this.data.fileId);
+    debugger;
```
Comment (mhuseinov) [inline alcs-frontend/src/app/features/application/documents/document-upload-dialog/document-upload-dialog.component.ts:152]: loos like a debug leftover

File: services/apps/alcs/src/alcs/application/application-submission/application-submission.service.ts [hunk 3]
```diff
@@ -17,6 +18,8 @@ export class ApplicationSubmissionService {
     private applicationSubmissionRepository: Repository<ApplicationSubmission>,
     @InjectRepository(ApplicationStatus)
     private applicationStatusRepository: Repository<ApplicationStatus>,
+    @InjectRepository(ApplicationParcel)
```
Comment (mhuseinov) [inline services/apps/alcs/src/alcs/application/application-submission/application-submission.service.ts:21]: I do not see references of this repo in the code, do you still need it?

File: alcs-frontend/src/app/features/application/documents/document-upload-dialog/document-upload-dialog.component.ts (hunk 2 above)
Comment (dhaselhan) [reply alcs-frontend/src/app/features/application/documents/document-upload-dialog/document-upload-dialog.component.ts]: Removed.

**Review by mhuseinov — APPROVED** [review]
🔥

File: services/apps/alcs/src/alcs/application/application-submission/application-submission.service.ts (hunk 3 above)
Comment (dhaselhan) [reply services/apps/alcs/src/alcs/application/application-submission/application-submission.service.ts]: Cleaned up, good catch

**Pull request merged**